File: server/src/test/java/de/zalando/zally/apireview/RestApiIgnoreRulesTest.java
package de.zalando.zally.apireview;

import de.zalando.zally.dto.ApiDefinitionResponse;
import org.junit.Test;
import org.springframework.test.context.TestPropertySource;

import java.io.IOException;

import static de.zalando.zally.util.ResourceUtilKt.readApiDefinition;
import static org.assertj.core.api.Assertions.assertThat;

@TestPropertySource(properties = "zally.ignoreRules=TestCheckAlwaysReport3MustViolations")
public class RestApiIgnoreRulesTest extends RestApiBaseTest {
    @Test
    public void testShouldNotReportViolationsIfIgnoreIsPresent() throws IOException {
        ApiDefinitionResponse response = sendApiDefinition(readApiDefinition("fixtures/openapi3_petstore_expanded.json"));
        assertThat(response.getViolations()).isEmpty();
    }
}

File: server/zally-server/src/test/kotlin/org/zalando/zally/apireview/RestApiIgnoreRulesTest.kt
package org.zalando.zally.apireview

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.test.context.TestPropertySource
import org.zalando.zally.util.readApiDefinition
import java.io.IOException

@TestPropertySource(properties = ["zally.ignoreRules=TestCheckAlwaysReport3MustViolations"])
class RestApiIgnoreRulesTest : RestApiBaseTest() {
    @Test
    @Throws(IOException::class)
    fun testShouldNotReportViolationsIfIgnoreIsPresent() {
        val response = sendApiDefinition(readApiDefinition("fixtures/openapi3_petstore_expanded.json"))
        assertThat(response.violations).isEmpty()
    }
}
